validation_aws/src/sbl_validation_processor/results_aggregator.py

```diff
@@ -56,40 +74,47 @@ def aggregate_validation_result(bucket, key):
         max_group_size = os.getenv("MAX_GROUP_SIZE", 200)
 
         if submission and submission.state not in [SubmissionState.SUBMISSION_ACCEPTED, SubmissionState.VALIDATION_EXPIRED, SubmissionState.SUBMISSION_UPLOAD_MALFORMED]:
-            aws_session = boto3.session.Session()
-            creds = aws_session.get_credentials()
-            storage_options = {
-                'aws_access_key_id': creds.access_key,
-                'aws_secret_access_key': creds.secret_key,
-                'session_token': creds.token,
-                'aws_region': 'us-east-1',
-            }
-            lf = pl.scan_parquet(f"s3://{bucket}/{key}", allow_missing_columns=True, storage_options=storage_options)
-            max_err_lf = lf.slice(0, max_errors + 1)
-            df = max_err_lf.collect()
+
+            file_paths, storage_options = get_parquet_paths(bucket, key)
+
+            #scan each result parquet into a lazyframe then diagonally concat so all columns are merged into the final lf.  Otherwise
+            #this will error if trying to scan a parquet directory and the parquets don't contain the same columns (particularly the
+            #field/value columns)
+            lazyframes = [pl.scan_parquet(file, allow_missing_columns=True, storage_options=storage_options) for file in file_paths]
+            lf = pl.LazyFrame()
+            if lazyframes:
+                lf = pl.concat(lazyframes, how="diagonal")
+            
+            #get the real total count of errors and warnings before truncating based on max error length
+            df = lf.collect()
             error_counts, warning_counts = get_scope_counts(df)
-            csv_df = pl.concat([df], how="diagonal")
-            csv_content = df_to_download(csv_df, warning_counts.total_count, error_counts.total_count, max_errors)
-            s3.put_object(Body=csv_content, Bucket=bucket, Key=validation_report_path)
+            #slice is start indice inclusive, so 0 to max_errors will return 1000000 errors (0-999999) if the 
+            #max_errors is 1000000 and there are more than that.  Adding +1 actually returns 
+            #max_errors + 1 which would be one more than the max_errors intended
```

Comment:
right, although maybe it doesn't matter for the new paradigm, but the intent was to make sure the downstream special case triggers, i.e. adding that extra line in the csv file, since at the max_errors, it doesn't trigger, but max_errors + 1 triggers the line to be added